wgengine/wgengine.go

```diff
@@ -45,31 +58,55 @@ func (rs *RouteSettings) OnlyRelevantParts() string {
 		rs.LocalAddr, rs.DNS, rs.DNSDomains, peers)
 }
 
+// Router is the TODO.
 type Router interface {
+	// Up brings the router up.
+	// TODO: more than once? after Close?
```

Comment:
Once. It might be possible to move this code into the New function that creates the router, but I'd have to dig through the initialization order to be sure.

I hate to think what happens after Close. let the object go.